### PR TITLE
Avoid extra loops on sorted data in the `__subgroup_bubble_sorter::sort`

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we avoid extra loops on sorted data in the `__subgroup_bubble_sorter::sort`,
which calls in sort algorithms on the device for every work item.

This is beneficial when the data is already sorted, which according to https://en.wikipedia.org/wiki/Sorting_algorithm is a common real-world case.

### CODE REVIEW

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h [hunk 1]
```diff
@@ -40,18 +40,39 @@ struct __subgroup_bubble_sorter
...
+
+        _IndexType __n = __end - __start;
+
+        switch (__n)
         {
-            for (std::uint32_t j = __start + 1; j < __start + __end - i; ++j)
+        case 0:
+        case 1:
+            break;
+        case 2:
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h:52]: I propose to ignore clang format at this place.

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h [hunk 2]
```diff
@@ -40,18 +40,39 @@ struct __subgroup_bubble_sorter
     void
     sort(const _StorageAcc& __storage_acc, _Compare __comp, std::uint32_t __start, std::uint32_t __end) const
     {
-        for (std::uint32_t i = __start; i < __end; ++i)
+        using _IndexType = std::make_signed_t<decltype(__end)>;
+
+        _IndexType __n = __end - __start;
+
+        switch (__n)
```
[the remark this answers is not in the document]
Comment (mmichel11) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h:47]: For large number of elements `switch` should perform better although I do not know in device code. However, with just 4 cases there's probably no difference.

In my opinion, unless there is a measurable performance difference between switch and if-else, then we should use if-else.

Comment (mmichel11) [inline include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h:47]: Could you add a small comment justifying these extra checks?

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h [hunk 3]
```diff
@@ -40,18 +40,39 @@ struct __subgroup_bubble_sorter
     void
     sort(const _StorageAcc& __storage_acc, _Compare __comp, std::uint32_t __start, std::uint32_t __end) const
     {
-        for (std::uint32_t i = __start; i < __end; ++i)
+        using _IndexType = std::make_signed_t<decltype(__end)>;
+
+        _IndexType __n = __end - __start;
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h]: Done

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h (hunk 2 above)
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h:45]: Done

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h [hunk 4]
```diff
@@ -40,18 +40,41 @@ struct __subgroup_bubble_sorter
...
+                auto& __second_item = __storage_acc[__start + 1];
                 if (__comp(__second_item, __first_item))
                 {
                     using std::swap;
                     swap(__first_item, __second_item);
                 }
             }
+            break;
+        default:            // The case when __end > __start + 2    three or more source data items require full bubble sort
+            do
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h:62]: Implementation from https://en.wikipedia.org/wiki/Bubble_sort

**Pull request closed without merging**